crates/autopilot/src/domain/settlement/observer.rs

```diff
@@ -11,21 +11,31 @@
 // used etc.
 
 use {
-    crate::{domain::settlement, infra},
+    crate::{
+        domain::settlement::{self, transaction::ContractTransactionAuthenticator},
```

Comment:
nit: mostly a matter of taste and code consistency but we learned that referring to types from their module can make the code in the module and at the call site readable.
For example the module is called `transaction`. If you only import the type one might want to call it `TransactionAuthenticator` but when you look at the source code it's often clear form the context what gets authenticated. To get the best of both worlds you could call the type `Authenticator` but always refer to it as `transaction::Authenticator` wherever you use it.